SFML/SFML.Net
# Configure .NET analyzer and code style settings

## Description

This PR switches code style setup from `.editorconfig` + `omnisharp.json` to `.editorconfig` + .NET's built-in analyzers. The main idea is to setup proper code rules that can also be enforced at compile time if needed.

I've split it into two commits: the one that configures the rules, and the one that fixes all of the warnings generated by said rules.

## How to test?

Grab the branch and build it, then go back a commit and check what warnings were being generated for the previous version of the code base.

## Breaking Changes

- Interfaces were changed to use the `I`-prefix in their name
- Some function parameters name were changed to `camelCase` instead of `PascalCase`
- DllImport names for SFML where changed to use a capital first letter, e.g. `SFML.Audio` instead of `SFML.audio`
- Transform uses capitalized members (`M1` instead of `m1`) that are internally accessible
